ui/app/src/components/dialogs/CreateDashboardDialog.tsx

```diff
@@ -92,32 +92,52 @@ interface DuplicationFormProps {
   onSuccess?: Dispatch<DashboardSelector | EphemeralDashboardInfo>;
 }
 
+/* TODO: Why does it receive an array of projects and not a single project?! */
 const DashboardDuplicationForm = (props: DuplicationFormProps): ReactElement => {
   const { projects, hideProjectSelect, onClose, onSuccess } = props;
 
-  const dashboardSchemaValidation = useDashboardValidationSchema();
+  const { schema: dashboardSchemaValidation, isSchemaLoading: isDashboardSchemaValidationLoading } =
+    useDashboardValidationSchema(projects[0]?.metadata.name);
 
   const dashboardForm = useForm<CreateDashboardValidationType>({
-    resolver: zodResolver(dashboardSchemaValidation),
+    resolver: dashboardSchemaValidation ? zodResolver(dashboardSchemaValidation) : undefined,
     mode: 'onBlur',
     defaultValues: { dashboardName: '', projectName: projects[0]?.metadata.name ?? '' },
   });
 
-  const processDashboardForm: SubmitHandler<CreateDashboardValidationType> = (data) => {
-    onClose();
-    if (onSuccess) {
-      onSuccess({ project: data.projectName, dashboard: data.dashboardName } as DashboardSelector);
-    }
-  };
+  const getProcessDashboardForm = useCallback((): SubmitHandler<CreateDashboardValidationType> => {
```

Comment:
why renaming the function here? After the changes I think `processDashboardForm` still makes more sense than `getProcessDashboardForm`